Fix remaining window-related test failures

@claude Please fix the remaining window-related test failures that are still failing in CI.

## Failing Tests

1. **SessionPersistenceService.browser.test.js** (6 failures)
   - Lines 15, 21, 39, 42, 51, 59
   - ReferenceError: window is not defined
   - AssertionError: expected false to be true
   - Error: Persistent storage not available

2. **ProfileService.test.ts** (1 failure)
   - Line 32
   - ReferenceError: window is not defined

3. **ProfileService.test.js** (1 failure)
   - Line 27
   - ReferenceError: window is not defined

4. **apiClient.test.ts** (1 failure)
   - Line 46
   - ReferenceError: window is not defined

5. **apiClient.test.js** (1 failure)
   - Line 38
   - ReferenceError: window is not defined

## Requirements

- Fix window/localStorage mocking for all failing tests
- Use vitest's happy-dom environment properly
- Ensure all tests pass with `npm run test:run`
- Once fixed, we can remove `continue-on-error: true` from .github/workflows/ci.yml

## Notes

The vitest.config.ts already uses happy-dom environment, but some tests still can't access window. These tests need proper browser environment setup or mocking.

--- a/src/lib/apiClient.test.js
+++ b/src/lib/apiClient.test.js
@@ -8,6 +8,8 @@
  * - Chat streaming
  * - Error handling
  * - Token management
+ *
+ * @vitest-environment happy-dom
  */
 import { describe, it, expect, beforeEach, afterEach, vi } from "vitest";
 import { ApiClient, ApiError } from "./apiClient.ts";

--- a/src/lib/apiClient.test.ts
+++ b/src/lib/apiClient.test.ts
@@ -8,6 +8,8 @@
  * - Chat streaming
  * - Error handling
  * - Token management
+ *
+ * @vitest-environment happy-dom
  */
 
 import { describe, it, expect, beforeEach, afterEach, vi } from "vitest";

--- a/src/services/ProfileService.test.js
+++ b/src/services/ProfileService.test.js
@@ -6,6 +6,8 @@
  * - Validation logic
  * - Error handling
  * - Memoization and caching
+ *
+ * @vitest-environment happy-dom
  */
 import { randomUUID } from "node:crypto";
 import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

--- a/src/services/ProfileService.test.ts
+++ b/src/services/ProfileService.test.ts
@@ -6,6 +6,8 @@
  * - Validation logic
  * - Error handling
  * - Memoization and caching
+ *
+ * @vitest-environment happy-dom
  */
 
 import { randomUUID } from "node:crypto";

--- a/src/services/SessionPersistenceService.browser.test.js
+++ b/src/services/SessionPersistenceService.browser.test.js
@@ -1,3 +1,5 @@
+// @vitest-environment happy-dom
+
 import { v4 as uuidv4 } from "uuid";
 import { beforeEach, describe, expect, it, vi } from "vitest";
 import { SessionPersistenceService } from "./SessionPersistenceService.ts";

